src/PublicApiGeneratorTests/NullableTests.cs

```diff
@@ -0,0 +1,459 @@
+        public System.DateTime? Birth { get; set; }
+        public float? Calc(double? first, decimal? second) { }
+        public System.Collections.Generic.List<System.Guid?> GetSecrets(System.Collections.Generic.Dictionary<int?, System.Collections.Generic.Dictionary<bool?, byte?>> data) { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_Generics()
+        {
+            AssertPublicApi<Generics>(
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class Generics
+    {
+           public Generics() { }
+           public System.Collections.Generic.List<string?> GetSecretData0() { }
+           public System.Collections.Generic.Dictionary<int, System.Collections.Generic.List<int?>> GetSecretData1() { }
+           public System.Collections.Generic.Dictionary<int, System.Collections.Generic.List<int?>?> GetSecretData2() { }
+           public System.Collections.Generic.Dictionary<int, System.Collections.Generic.List<System.Collections.Generic.KeyValuePair<string?, System.Collections.Generic.List<int>?>>> GetSecretData3(System.Collections.Generic.Dictionary<int?, System.Collections.Generic.List<System.Collections.Generic.KeyValuePair<string, System.Collections.Generic.List<int?>>>>? value) { }
+           public System.Collections.Generic.Dictionary<int?, string>? GetSecretData4(System.Collections.Generic.Dictionary<int?, string>? value) { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_Structs()
+        {
+            AssertPublicApi<Structs>(
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class Structs
+    {
+           public System.Collections.Generic.KeyValuePair<string?, int?> field;
+           public Structs() { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_Tuples()
+        {
+            AssertPublicApi<Tuples>(
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class Tuples
+    {
+           public Tuples() { }
+           public System.Tuple<string, string?, int, int?> Tuple1(System.Tuple<string, string?, int, int?>? tuple) { }
+           public System.ValueTuple<string, string?, int, int?> Tuple2(System.ValueTuple<string, string?, int, int?>? tuple) { }
+    }
+}");
+        }
+
+        [Fact]
+        public void Should_Annotate_Constraints()
+        {
+            AssertPublicApi<Constraints>(
+@"namespace PublicApiGeneratorTests.Examples
+{
+    public class Constraints
+    {
+           public Constraints() { }
+           public void Print1<T>(T val)
+               where T : class { }
+           public void Print2<T>(T val)
+               where T : class? { }
+           public static void Print3<T>()
+                where T : System.IO.Stream { }
+           public static void Print4<T>()
+                where T : System.IDisposable { }
+    }
+}");
+        }
+
+        [Fact(Skip = "There is a problem with these tests - the attribute says 1 instead of 2")]
```

Comment:
Help is appreciated.

Reply:
I pinged Immo over twitter. Maybe he has some input

Reply:
In fact, we found a place where these attributes are stored. And it seems that they are hard to get.